applications/API/serializers.py

```
@@ -0,0 +1,554 @@
+        read_only_fields = ["reference_id", "status", "added_by", "added_on"]
+
+    def find_category(self, obj) -> str:
+        """
+        Resolve and format the category name efficiently, handling either a raw
+        key/name string or an InsultCategory FK instance on the model.
+        """
+        try:
+            category_value = obj.category
+
+            # If FK instance is loaded, short-circuit without hitting validation branches
+            from applications.API.models import InsultCategory as _IC
```

Pylint (3.3.7)

[warning] Reimport 'InsultCategory' (imported line 18)

(W0404)